# The Python Challenge 

## Level 16 - let me get this straight

### Url: http://huge:file@www.pythonchallenge.com/pc/return/mozart.html

习惯性地先把图片整下来……等会，这图怎么是.gif？？

但是这个gif就这一帧，所以还是按普通图片来处理比较好。

根据标题，我们需要利用图里这些粉色（紫色）的短线。

肉眼可以看到，这些紫线的规格是一样的，都是两端为白点，中间夹着长度一致的紫色线条，接下来想办法获取这个紫线的数据。

最顶上那一行的紫线在后半部分，那么简单搜索一下300到600像素：

In [ ]:
for x in range(300, 600):
    print(im.getpixel((x, 0)), end=' ')


16 60 96 95 60 88 96 88 89 54 60 16 96 23 88 48 24 82 16 17 88 17 60 59 89 46 54 9 90 23 88 60 96 23 47 60 88 22 11 88 120 100 22 18 47 46 60 82 96 88 95 60 58 84 83 90 24 95 82 9 59 54 88 18 132 96 60 17 96 58 59 59 84 9 48 16 89 88 132 97 59 88 95 125 96 88 96 94 22 96 89 60 23 60 58 60 88 24 23 82 96 46 84 47 82 60 11 95 60 22 132 54 60 84 58 95 46 11 60 95 11 17 18 60 16 59 60 60 249 195 195 195 195 195 252 88 48 96 47 88 24 17 96 94 60 96 89 16 54 17 90 102 11 60 84 60 60 89 16 22 54 23 60 48 88 17 24 23 82 22 48 94 96 18 119 102 54 18 24 82 22 48 46 90 23 54 22 59 47 90 24 16 22 53 102 94 60 94 18 18 58 48 18 59 23 60 17 17 59 23 95 95 88 58 48 101 48 24 95 60 60 59 17 59 23 83 18 47 59 17 95 60 17 17 60 18 60 131 23 96 53 16 133 82 23 54 94 60 90 88 19 90 16 96 60 60 16 95 91 60 18 90 24 18 124 23 54 90 94 54 96 91 95 18 18 48 59 95 95 88 18 119 60 23 91 95 59 22 60 54 94 11 47 16 60 60 24 96 90 18 

肉眼观察，找到中间出现了 [249,195,195,195,195,195,252] 的序列，可见这个紫色的值为195，另外值得注意的是两端的白点不是固定255，这是一个小坑。

接下来应该就是根据紫线来重排每一行的像素，令紫线为一行的开头，紫线前面的部分切下来接到末尾。

## Next Level:

### Url: http://huge:file@www.pythonchallenge.com/pc/return/romance.html